widget/embedded/src/components/ConfirmWalletsModal/WalletList.tsx

```diff
@@ -73,55 +62,74 @@ export function WalletList(props: PropTypes) {
   const shouldShowMoreWallets = limit && numberOfSupportedWallets - limit > 0;
 
   const addExperimentalChain = async (wallet: Wallet) => {
-    setExperimentalChainWallet(null);
     setShowExperimentalChainModal(false);
+    setAddingExperimentalChainStatus('in-progress');
     await connect(wallet.walletType, wallet.chain);
+    setAddingExperimentalChainStatus('completed');
     selectWallet({
       walletType: wallet.walletType,
       chain,
       address: wallet.address ?? '',
     });
   };
 
+  const sortedWalletsBasedOnSelectedStatus = list.slice(0, 1).concat(
+    list.slice(1).sort((walletA, walletB) => {
+      const walletASelected = isSelected(walletA.type, chain) ? 1 : 0;
```

Comment:
Is it possible to not change the sort when something is selected from show-more wallets but we switch to another wallet. (We could skip it for now if it completely changes your code logic)